src/actions/index.js

```diff
@@ -152,58 +153,59 @@ export const clearAll =
     }
 
 export const tDoRenameVisualization =
-    ({ name, description }) =>
+    ({ name, description }, onRenameComplete) =>
     async (dispatch, getState, engine) => {
-        const onSuccess = (res) => {
-            if (res.status === 'OK' && res.response.uid) {
-                const state = getState()
-
-                const visualization = sGetVisualization(state)
-                const current = sGetCurrent(state)
-
-                const updatedVisualization = { ...visualization }
-                const updatedCurrent = { ...current }
-
-                if (name) {
-                    updatedVisualization.name = updatedCurrent.name = name
-                }
+        const onSuccess = (updatedVis) => {
+            const state = getState()
 
-                updatedVisualization.description = updatedCurrent.description =
-                    description
+            const visualization = sGetVisualization(state)
+            const current = sGetCurrent(state)
 
-                dispatch(
-                    fromVisualization.acSetVisualization(updatedVisualization)
-                )
+            const updatedVisualization = { ...visualization, ...updatedVis }
```

Comment:
Here we merge the "old" visualization with the new name/description.
Is this correct?
`visualization` usually reflects what is stored in the DB, and although it can get out of sync, with the changes in the PR we do fetch the "latest" from the DB, so I'm wondering if this merge should use the "latest" instead of whatever we had in Redux.

Reply:
Yes I intentionally only update the name/desc in the redux `visualization` prop. This was to limit the scope of this PR. I wasn't sure what side effects there would be to updating the whole object.